pkg/macaroon/macaroon_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package macaroon_test
+
+import (
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+	"storj.io/storj/pkg/macaroon"
+)
+
+func TestNilMacaroon(t *testing.T) {
```

Comment:
I would love to see a fuzzing test, because it's easy to get parsing wrong.

Reply:
one thought i had looking at this - instead of writing all of this varint serialization by hand, etc, this is... exactly what protobufs are good for. it probably departs from any standards, but we could just do

```
message Macaroon {
   bytes head = 1;
   repeated bytes caveat = 2;
   bytes tail = 3;
}
```
and all of the parsing/serialization code would be written?

Reply:
I would rather follow a standard over pb

Reply:
well, "standard". the paper didn't propose a wire format, but perhaps working with https://github.com/rescrv/libmacaroons is good?

Reply:
Yeah, looks like there's no great standard https://news.ycombinator.com/item?id=16073756.

Reply:
https://github.com/rescrv/libmacaroons/blob/master/doc/format.txt is emerging as a standard that other libraries support.